Backend/ServiceLayer/Services/ResetService.cs

```diff
@@ -0,0 +1,229 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using DataAccessLayer.Database;
+using DataAccessLayer.Models;
+using DataAccessLayer.Repositories;
+using MimeKit;
+using MailKit;
+using MailKit.Net.Smtp;
+
+
+namespace ServiceLayer.Services
+{
+    public class ResetService
+    {
+
+        #region Constants for the reset service
+        //Variable for how long the token is supposed to be live
+        private const double TimeToExpire = 5;
+        //Constant for the password reset email subject line
+        private const string resetPasswordSubjectString = "KFC SSO Password Reset";
+        //URL to append the reset ID to. Sent to the user in the email body.
+        private const string resetControllerURL = "kfcsso.com/api/reset/resetpassword/?id=";
+
+        #endregion
+
+        private ResetRepository _resetRepo;
+        public ResetService()
+        {
+            _resetRepo = new ResetRepository(new DatabaseContext());
+        }
+
+        //Function to create token in the system
+        public string CreateResetID()
+        {
+            
+            RNGCryptoServiceProvider rngCsp = new RNGCryptoServiceProvider();
+            //List used to hold the two random numbers that will be generated
+            List<string> listOfInts = new List<string>();
+            //While loop to continue generating the numbers until satisfies length requirement
+            while (listOfInts.Count < 2)
+            {
+                byte[] randomNumber = new byte[8];
+                rngCsp.GetBytes(randomNumber);
+                int result = BitConverter.ToInt32(randomNumber, 0);
+                //If the number that was generated is negative, make it positive
+                if (result < 0)
+                {
+                    result = result * -1;
+                }
+                //Convert the number to a string
+                string resultToString = result.ToString();
+                //Add the number to the list of random numbers generated
+                listOfInts.Add(resultToString);
+            }
+            //Concatenate the two numbers generated to have a resetID that is minimum of 18 characters long
+            string resetID = listOfInts[0] + listOfInts[1];
+
+            return resetID;
+        }
+
+        public string CreateResetURL(string resetID)
+        {
+            string resetURL = resetControllerURL + resetID;
+            return resetURL;
+        }
+
+        //Function to add the token to the database
+        public void AddResetIDToDB(string userName, string resetID)
+        {
```

Comment:
Service classes should reflect changes in the repository classes

PasswordReset createPasswordReset(GUID userID, DatabaseContext _db) {
    PasswordReset newPasswordReset = {...}
    _passwordResetRepo.createPasswordReset(newPasswordReset, _db);
}

Need services classes for all CRUD operations and useful logic operaitons such as 
boolean isPasswordResetExpired(PasswordReset passwordReset) //checks if given reset has already passed expiration. You can remove redundant methods that the new CRUD methods will cover such as getSecurityQuestions and lockResetID.